### PR TITLE
Update Time Sheet for compensation date

### PR DESCRIPTION
**Pull request by daolq3012**
Update Time Sheet for compensation date

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/data/model/UserTimeSheet.java [hunk 1]
```diff
@@ -86,4 +91,11 @@ public void setTotalFining(float totalFining) {
     public void setTimeSheetDates(List<TimeSheetDate> timeSheetDates) {
         mTimeSheetDates = timeSheetDates;
     }
+
+    public int getCutOffDate() {
+        if (StringUtils.isBlank(mStartDate)) {
+            return -1;
+        }
+        return DateTimeUtils.convertStringToDate(mStartDate, DateTimeUtils.DATE_FORMAT_YYYY_MM_DD_2).getDate() - 1;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/data/model/UserTimeSheet.java]: ERROR: Line is longer than 100 characters (found 115).

File: app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetModule.java [hunk 2]
```diff
@@ -30,10 +29,9 @@ public TimeSheetModule(@NonNull Fragment fragment) {
 
     @FragmentScope
     @Provides
-    public TimeSheetContract.ViewModel provideViewModel(Context context,
-            TimeSheetContract.Presenter presenter, Navigator navigator,
+    public TimeSheetContract.ViewModel provideViewModel(TimeSheetContract.Presenter presenter, Navigator navigator,
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetModule.java]: ERROR: Line is longer than 100 characters (found 115).

File: app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java [hunk 3]
```diff
@@ -86,6 +83,7 @@ public void onGetTimeSheetError(BaseException throwable) {
 
     @Override
     public void onGetTimeSheetSuccess(UserTimeSheet userTimeSheet) {
+        setCutOffDate(userTimeSheet.getCutOffDate() == -1 ? mCutOffDate : userTimeSheet.getCutOffDate());
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java]: ERROR: Line is longer than 100 characters (found 105).

File: app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java [hunk 4]
```diff
@@ -86,7 +84,10 @@ public void onGetTimeSheetError(BaseException throwable) {
 
     @Override
     public void onGetTimeSheetSuccess(UserTimeSheet userTimeSheet) {
+        setCutOffDate(userTimeSheet.getCutOffDate() == -1 ?
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java]: WARNING: '?' should be on a new line.

File: app/src/main/java/com/framgia/wsm/utils/binding/BindingUtils.java [hunk 5]
```diff
@@ -176,10 +176,10 @@ public static void setBackground(CardView cardView, String color) {
         cardView.setCardBackgroundColor(Color.parseColor(color));
     }
 
-    @BindingAdapter({ "timeSheetDates", "month", "year", "cutOffDate" })
+    @BindingAdapter({ "timeSheetDates", "month", "year", "cutOffDate", "monthHaveCompensation" })
     public static void setTimeSheetDates(TimeSheetView timeSheetView,
-            List<TimeSheetDate> timeSheetDates, int month, int year, int cutOffDate) {
-        timeSheetView.setTime(month, year, cutOffDate);
+            List<TimeSheetDate> timeSheetDates, int month, int year, int cutOffDate, boolean isCompensationMonth) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/utils/binding/BindingUtils.java]: ERROR: Line is longer than 100 characters (found 115).

File: app/src/main/java/com/framgia/wsm/widget/timesheet/TimeSheetView.java [hunk 6]
```diff
@@ -607,7 +598,7 @@ public void setFirstDateOfMonth(int month, int year) {
         mFirstDateOfMonth = mCutOffDate + 1;
     }
 
-    public void setTime(int month, int year, int cutOffDate) {
+    public void setTime(int month, int year, int cutOffDate, boolean isCompensationMonth) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/widget/timesheet/TimeSheetView.java]: WARNING: Method length is 69 lines (max allowed is 60).

File: app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java [hunk 7]
```diff
@@ -334,8 +335,18 @@ public int getCutOffDate() {
         return mCutOffDate;
     }
 
+    @Bindable
+    public boolean getMonthHaveCompensation() {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java]: WARNING: [BooleanGetMethodName] A getX() method which returns a boolean should be named isX()
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#BooleanGetMethodName

**Comment by daolq3012** [comment]
OK!

**Pull request merged**